From 34fad5aeb307fed09afba3d3e0e9af9af4ca935e Mon Sep 17 00:00:00 2001
From: Daniel Emminizer <daniel.m.emminizer.civ@us.navy.mil>
Date: Wed, 30 Apr 2025 14:57:11 +0000
Subject: [PATCH 29/29] simUtil: DbConfigurationFile's use of setColorFilter is
 whitelisted from deprecation warnings

**JIRA Issue:** SIM-18211

**Description:** On GCC, ignore deprecation warnings for setColorFilter, for alpha color filter, used in cloud processing on legacy Db configuration .txt files.

**Notes:** Eventually this can migrate to a layer shader.

**Testing Performed:** Builds without warning on MSVC 2022 and gcc Linux.
---
 SDK/simUtil/DbConfigurationFile.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/SDK/simUtil/DbConfigurationFile.cpp b/SDK/simUtil/DbConfigurationFile.cpp
index 8d606eba..d3e20b97 100644
--- a/SDK/simUtil/DbConfigurationFile.cpp
+++ b/SDK/simUtil/DbConfigurationFile.cpp
@@ -452,7 +452,15 @@ void DbConfigurationFile::parseCloudLayers_(const std::vector<std::string>& toke
       if (clearVal < opaqueVal && (opaqueVal != 1.0f || clearVal != 0.0f) && simVis::AlphaColorFilter::isSupported())
       {
         simVis::AlphaColorFilter* filter = new simVis::AlphaColorFilter();
+#ifdef __GNUC__
+        // continue using deprecated addColorFilter()
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
         imageLayer->addColorFilter(filter);
+#pragma GCC diagnostic pop
+#else
+        imageLayer->addColorFilter(filter);
+#endif
         osg::Vec2f values(clearVal, opaqueVal);
         filter->setAlphaOffset(values);
       }
-- 
2.47.1.windows.2

